hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -600,10 +613,10 @@ public void deleteFilesystem(TracingContext tracingContext)
     }
   }
 
-  public OutputStream createFile(final Path path,
+  public OutputStream createFile(final Path path, final boolean isFile,
       final FileSystem.Statistics statistics, final boolean overwrite,
       final FsPermission permission, final FsPermission umask,
-      TracingContext tracingContext) throws IOException {
+      TracingContext tracingContext, HashMap<String, String> metadata) throws IOException {
```

Comment:
if its known its a directory marker, why would metadata be needed ?

Reply:
Metadata is needed for sending metadata during put blob